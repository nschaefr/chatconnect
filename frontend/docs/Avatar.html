<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Avatar</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Avatar</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Avatar<span class="signature">(username, userId)</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="Avatar"><span class="type-signature"></span>new Avatar<span class="signature">(username, userId)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>A component for the avatar from the User</p>
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>username</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>Username</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>userId</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>User id</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_chat-page_avatar.jsx.html">components/chat-page/avatar.jsx</a>, <a href="components_chat-page_avatar.jsx.html#line8">line 8</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="~getColorByHash"><span class="type-signature">(inner) </span>getColorByHash<span class="signature">(hash)</span><span class="type-signature"> &rarr; {array}</span></h4>
    

    











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>hash</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>hash to generate user color</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_chat-page_avatar.jsx.html">components/chat-page/avatar.jsx</a>, <a href="components_chat-page_avatar.jsx.html#line32">line 32</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>colors - returns array with colors</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">array</span>


    </dd>
</dl>

    





        
            

    

    
    <h4 class="name" id="~hashCode"><span class="type-signature">(inner) </span>hashCode<span class="signature">(str)</span><span class="type-signature"> &rarr; {string}</span></h4>
    

    











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>str</code></td>
            

            <td class="type">
            
                
<span class="param-type">string</span>


            
            </td>

            

            

            <td class="description last"><p>userId to generate hash for color for user</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_chat-page_avatar.jsx.html">components/chat-page/avatar.jsx</a>, <a href="components_chat-page_avatar.jsx.html#line19">line 19</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>hash - hash to generate user color</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">string</span>


    </dd>
</dl>

    





        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="Avatar.html">Avatar</a></li><li><a href="Chat.html">Chat</a></li><li><a href="Contact.html">Contact</a></li><li><a href="Routes.html">Routes</a></li><li><a href="SignuploginForm.html">SignuploginForm</a></li></ul><h3>Global</h3><ul><li><a href="global.html#App">App</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 4.0.2</a> on Thu Sep 07 2023 17:10:02 GMT+0200 (Central European Summer Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>