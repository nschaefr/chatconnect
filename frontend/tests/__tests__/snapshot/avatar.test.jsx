import React from "react";
import { render } from "@testing-library/react";
import Avatar from "../../../src/components/chat-page/avatar";

describe("avatar snapshots", () => {
  test("render avatar component", () => {
    const { container } = render(<Avatar username="testuser" userId={"123"} />);

    expect(container).toMatchSnapshot();
  });
});
