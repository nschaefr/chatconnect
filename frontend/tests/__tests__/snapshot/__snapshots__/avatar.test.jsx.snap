// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avatar snapshots render avatar component 1`] = `
<div>
  <div
    data-testid="avatar"
    style="background-color: rgb(153, 207, 201); width: 35px; height: 35px; display: flex; border-radius: 50%; align-items: center; justify-content: center;"
  >
    <div
      style="font-family: ExtraBold; color: white; font-size: 15px;"
    >
      T
    </div>
  </div>
</div>
`;
